src/runtime/hexagon/hexagon_device_target.h

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright 2019 Qualcomm Innovation Center, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE‐2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This Contribution is being provided by Qualcomm Technologies, Inc.,
+ * a Delaware corporation, or its subsidiary Qualcomm Innovation Center, Inc.,
+ * a California corporation, under certain additional terms and conditions
+ * pursuant to Section 5 of the Apache 2.0 license.  In this regard, with
+ * respect to this Contribution, the term "Work" in Section 1 of the
+ * Apache 2.0 license means only the specific subdirectory within the TVM repo
+ * (currently at https://github.com/dmlc/tvm) to which this Contribution is
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ */
+#ifndef TVM_RUNTIME_HEXAGON_HEXAGON_DEVICE_TARGET_H_
+#define TVM_RUNTIME_HEXAGON_HEXAGON_DEVICE_TARGET_H_
+#ifdef __ANDROID__
+#include <android/log.h>
+
+#include <map>
+#include <string>
+
+#include "AEEStdErr.h"
+#include "fastrpc/tvm_hexagon_remote.h"
+#include "hexagon_module.h"
+#include "rpcmem.h"
+
+#define RPCMEM_HEAP 25
+#define LOG_TVM __android_log_print
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+class HexagonTarget : public tvm::runtime::hexagon::Device {
```

Comment:
do not need header as you can implement this in a cc file and return a std::shared_ptr<HexagonDevice>

Reply:
I suppose I could have a function "ConstructHexagonTarget" returning unique_ptr<Device>, and same for HexagonSimulator.  They could be declared in hexagon_module, and these headers could be moved into their .cc files.  (There is no need for std::shared_ptr here.)

Reply:
Done (removed the header).